expansion/src/test/java/no/unit/nva/expansion/model/ExpandedDataEntryTest.java

```diff
@@ -127,77 +134,128 @@ public void setup() {
         this.resourceService = getResourceServiceBuilder().build();
         this.messageService = getMessageService();
         this.ticketService = getTicketService();
-        this.uriRetriever = mock(UriRetriever.class);
+        this.uriRetriever = FakeUriRetriever.newInstance();
         this.resourceExpansionService = new ResourceExpansionServiceImpl(resourceService, ticketService);
     }
 
     @ParameterizedTest()
     @MethodSource("importCandidateContextTypeProvider")
-    void shouldExpandImportCandidateSuccessfully(PublicationContext publicationContext) {
+    void shouldExpandImportCandidateSuccessfully(PublicationContext publicationContext) throws JsonProcessingException {
         var importCandidate = randomImportCandidate(publicationContext);
-        importCandidate.getEntityDescription().getContributors().stream()
-            .map(Contributor::getAffiliations)
-            .flatMap(List::stream)
-            .filter(Organization.class::isInstance)
-            .map(Organization.class::cast)
-            .forEach(this::mockOrganizations);
+        FakeUriResponse.setupFakeForType(importCandidate, uriRetriever);
+
         var expandedImportCandidate = ExpandedImportCandidate.fromImportCandidate(importCandidate, uriRetriever);
         assertThat(importCandidate.getIdentifier(), is(equalTo(expandedImportCandidate.identifyExpandedEntry())));
         this.resourceExpansionService = new ResourceExpansionServiceImpl(resourceService, ticketService);
     }
 
-    @ParameterizedTest()
-    @MethodSource("importCandidateContextTypeProvider")
-    void shouldExpandImportCandidateSuccessfullyWhenBadResponseFromCustomerApi(PublicationContext publicationContext) {
-        var importCandidate = randomImportCandidate(publicationContext);
+
+    @Test
+    void shouldUseCustomCristinApiCallWhenCristinApiCallIsUsed() throws JsonProcessingException {
```

Comment:
This seems very specific? Are we actually testing `shouldExpandImportCandidateCristinOrgWhenAffiliatedWithNvaCustomer`?

Reply:
Or something like that?